apollo-router/src/configuration/mod.rs

```diff
@@ -610,6 +610,11 @@ pub(crate) struct Supergraph {
     #[serde(rename = "experimental_reuse_query_fragments")]
     pub(crate) reuse_query_fragments: Option<bool>,
 
+    /// Enable QP generation of fragments for subgraph requests
+    /// Default: false
+    #[serde(rename = "generate_query_fragments")]
+    pub(crate) generate_query_fragments: Option<bool>,
```

Comment:
Both `reuse_query_fragments` and `generate_query_fragments` should not be wrapped in an `Option` instead set the default in the implementation of `Default`

Reply:
I might need a bit of help with what you mean here.

I think `reuse_query_fragments` is intentionally an `Option` for historical reasons. IIUC, the router defers to QP to set the default based on its version, though perhaps that's not really an issue since versions are pretty tightly coupled?

I removed the `Option` from `generate_query_fragments` anyway. But I'm not sure if the change in https://github.com/apollographql/router/pull/4885/commits/4c5ef94cda795b3055513ef9dab21853608f2192 is quite what you're after - there's nothing really _in_ `impl Default for Supergraph`, should I be adding something there?

Something like this perhaps?
```rust
impl Default for Supergraph {
    fn default() -> Self {
        Self::builder()
            .generate_query_fragments(false)
            .reuse_query_fragments(true)
            .build()
    }
}
```

Reply:
You got it correct. The `Default` impl delegates to the new contructor.